### ISSUE TITLE
az login into the AzureStackUser cloud fails

### ISSUE THREAD

**Issue by mclagett** [Azure/azure-cli#14969]
## Describe the bug

**Command Name**
`az login`

**Errors:**
```
'NoneType' object is not iterable
Traceback (most recent call last):
Local\Temp\pip-unpacked-wheel-cqhwq_hi\knack\cli.py, ln 215, in invoke
azure\cli\core\commands\__init__.py, ln 654, in execute
azure\cli\core\commands\__init__.py, ln 718, in _run_jobs_serially
azure\cli\core\commands\__init__.py, ln 711, in _run_job
AppData\Local\Temp\pip-unpacked-wheel-b47qj8aq\six.py, ln 703, in reraise
azure\cli\core\commands\__init__.py, ln 688, in _run_job
azure\cli\core\commands\__init__.py, ln 325, in __call__
pip-unpacked-wheel-_c27a1q9\azure\cli\core\__init__.py, ln 782, in default_command_handler
azure\cli\command_modules\profile\custom.py, ln 159, in login
pip-unpacked-wheel-_c27a1q9\azure\cli\core\_profile.py, ln 240, in find_subscriptions_on_login
pip-unpacked-wheel-_c27a1q9\azure\cli\core\_profile.py, ln 276, in _normalize_properties
TypeError: 'NoneType' object is not iterable
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az login -u {azurestack@mclagetthotmail.onmicrosoft.com} --tenant {mclagetthotmail.onmicrosoft.com}`

## Expected Behavior

Expected to log on to the active cloud, which is the AzureStackUser cloud that was successfully added with a previous az cloud register command and made active with the az cloud set command.  (The AzureStackUser cloud is listed when az cloud list is issued and is marked with IsActive: true.)
 
## Environment Summary
```
Windows-10-10.0.17763-SP0
Python 3.6.8
Installer: MSI

azure-cli 2.11.0

```
## Additional Context

The command was issued on a client machine with the Server 2019 OS installed that is connected via a VPN to the Azure Stack SDK host machine (which is also running Server 2019).  This VPN is working from other respects, as I am able to load the Azure Stack SDK user and admin portals through it as well as SSH into Azure Stack vms via their public ip addresses.

**Comment by yonzhan** [comment]
add to S175

**Comment by jiasli** [comment]
This is because you used an incorrect API version/profile on Azure Stack and triggered an Azure Stack bug https://github.com/Azure/azure-rest-api-specs/issues/9567.

I have already made a better error message in https://github.com/Azure/azure-cli/pull/15550. Per https://github.com/Azure/azure-cli/commit/24fe6f6a5843518639f0f3a16a9b7a3e64e103e2, the change is included since `2.15.0`.
